js/data/buffer.js

```diff
@@ -170,18 +169,16 @@ Buffer.prototype._createPushMethod = function() {
         body += '\nvar ' + offsetId + ' = (offset + ' + attribute.offset + ') / ' + attribute.type.size + ';\n';
 
         for (var j = 0; j < attribute.components; j++) {
-            var valueId = 'value' + i + '_' + j;
+            var rvalue = 'value.' + attribute.name + '[' + j + ']';
             var lvalue = 'this.views.' + attribute.type.name + '[' + offsetId + ' + ' + j + ']';
 
-            body += lvalue + ' = ' + valueId + ';\n';
-
-            argNames.push(valueId);
+            body += lvalue + ' = ' + rvalue + ';\n';
         }
     }
 
     body += '\nreturn index;\n';
 
-    this.push = new Function(argNames, body);
+    this.push = new Function('value', body);
```

Comment:
Using an object here instead of raw arguments has a measurable performance impact. We are going to have to take a performance hit *somewhere* in order to allow us to selectively add / not add attributes. I think this a good place to do it.